In [1]:
i = 1
while True:
    if i%3 == 0:
        break
    print(i)
 
    i+= 1

1
2


In [2]:
i = 1
while True:
    if i%3 == 0:
        break
    print(i)
 
    i + = 1 #for space it will give error

SyntaxError: invalid syntax (3946220663.py, line 7)

In [3]:
print(2**(3**2))

512


In [4]:
print((2**3)**2)

64


In [5]:
print(2**3**2)

512


The 3rd(print(2**3**2)) expression is evaluated as 2**(3**2). This is because the associativity of ** operator is from right to left. Hence the result of the third expression is 512.

In [6]:
l=[1, 0, 2, 0, 'hello', '', []]
list(filter(bool, l))

[1, 2, 'hello']

The code shown above returns a new list containing only those elements of the list l which do not amount to zero. Hence the output is: [1, 2, ‘hello’].

In [11]:
def f(x):
    def f1(*args, **kwargs):
           print("Sanfoundry")
           return x(*args, **kwargs)
    return f1

In [9]:
min(max(False,-3,-4), 2,7)

False

The function max() is being used to find the maximum value from among -3, -4 and false. Since false amounts to the value zero, hence we are left with min(0, 2, 7) Hence the output is 0 (false).

In [10]:
x=56.236
print("%.2f"%x)

56.24


The expression shown above rounds off the given number to the number of decimal places specified. Since the expression given specifies rounding off to two decimal places, the output of this expression will be 56.24. Had the value been x=56.234 (last digit being any number less than 5), the output would have been 56.23.

In [12]:
x = 'abcd'
for i in x:
    print(i.upper())

A
B
C
D


In [13]:
for i in [1, 2, 3, 4][::-1]:
    print (i) 
    #[::-1] reverses the list.

4
3
2
1


In [16]:
#dt
class tester:
    def __init__(self, id):
        self.id = str(id)
        id="224"
 
    temp =tester(12)
    print(temp.id)

NameError: name 'tester' is not defined